src/pages/public_registration/public_files/FileUploaderInfo.tsx

```diff
@@ -0,0 +1,68 @@
+import { Skeleton, Typography, TypographyProps } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { useFetchOrganization } from '../../../api/hooks/useFetchOrganization';
+import { useFetchPerson } from '../../../api/hooks/useFetchPerson';
+import { useFetchUserQuery } from '../../../api/hooks/useFetchUserQuery';
+import { ImportUploadDetails, UserUploadDetails } from '../../../types/associatedArtifact.types';
+import { getFullCristinName } from '../../../utils/user-helpers';
+
+interface PendingFileDetailsProps extends TypographyProps {
+  uploadDetails?: UserUploadDetails | ImportUploadDetails;
+}
+
+export const FileUploaderInfo = ({ uploadDetails, ...typographyProps }: PendingFileDetailsProps) => {
+  return uploadDetails?.type === 'UserUploadDetails' ? (
+    <UserUploaderInfo uploadDetails={uploadDetails} {...typographyProps} />
+  ) : uploadDetails?.type === 'ImportUploadDetails' ? (
+    <ImportUploaderInfo uploadDetails={uploadDetails} {...typographyProps} />
+  ) : null;
+};
+
+interface UserUploaderInfoProps extends TypographyProps {
+  uploadDetails: UserUploadDetails;
+}
+
+const UserUploaderInfo = ({ uploadDetails, ...typographyProps }: UserUploaderInfoProps) => {
+  const { t } = useTranslation();
+  const userQuery = useFetchUserQuery(uploadDetails.uploadedBy, { staleTime: Infinity, gcTime: 1_800_000 });
+
+  const personQuery = useFetchPerson(userQuery.data?.cristinId ?? '', { staleTime: Infinity, gcTime: 1_800_000 });
+  const organizationQuery = useFetchOrganization(userQuery.data?.institutionCristinId ?? '');
+
+  const fullName = getFullCristinName(personQuery.data?.names);
+  const organizationName = organizationQuery.data?.acronym;
+  const uploader = [fullName, organizationName].filter(Boolean).join(', ');
+
+  const isPending =
+    (userQuery.isPending && !userQuery.isError) ||
+    (userQuery.data && (personQuery.isFetching || organizationQuery.isFetching));
+
+  if (!isPending && !uploader) {
+    return null;
+  }
+
+  return (
+    <div aria-busy={isPending} aria-live="polite">
```

Comment:
Har du fått testet dette med skjermleser? Jeg får ikke noe info med VoiceOver, men det kan være brukerfeil 😅

Reply:
Nei, bare basert på docs: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-busy
Men godt poeng! usikker på hvordan man best verifiserer det også, for effekten skal vel bare være at den _ikke_ leser opp innholdet når det er skeleton, men leser opp navnet når det er klart. Men hva er sjansen for at brukeren har fokset der akkurat da og faktisk skal høre det 🤔